regserver/mock/server.go

```diff
@@ -10,12 +10,13 @@ import (
 	"github.com/qri-io/registry/regserver/handlers"
 )
 
-var nilSearch registry.NilSearch
+// var nilSearch registry.NilSearch
+var mockSearch = &registry.MockSearch{}
```

Comment:
Looks like this is being created without a `Datasets` implementation. we're sure this works?

Reply:
I guess this function wasn't being directly tested but I have updated to 
```go
func NewMockServer() (*regclient.Client, *httptest.Server) {
	memDs := registry.NewMemDatasets()
	s := httptest.NewServer(handlers.NewRoutes(handlers.NewNoopProtector(), registry.NewMemProfiles(), memDs, &registry.MockSearch{memDs}))
	c := regclient.NewClient(&regclient.Config{Location: s.URL})
	return c, s
}
```
(matching the regclient tests)